compiler/AST/ForLoop.cpp

```diff
@@ -200,6 +200,10 @@ BlockStmt* ForLoop::doBuildForLoop(Expr*      indices,
     loop->orderIndependentSet(true);
   }
 
+  if (!isForeach || isLoweredForall || isForExpr) {
+    loop->exemptFromImplicitIntents();
```

Comment:
It wasn't clear to me what this exactly is from reading the PR message. We mark some `for` loops as order-independent (`for` loops in iterators that just yield, for example). Is this a protection against such loops?

Reply:
In any case, it might be good to clarify that in the PR message as well.

Reply:
I want my implicit intents stuff to only kick on for user-written 'foreach' loops.  At first I thought `isOrderIndependentSet` would be synonymous with that but as you mentioned we lower some `for` loops as orderindependent (hence why I exempt `!isForeach`).

Apparantly, `isForEach` can be true for other types of loops

Namely, I don't exempt `isForExpr` then I get a test (`test/library/packages/ParallelIO/lines.chpl`) that errors out with this (saying you need to get a ref intent for `rng`):

`const x = [i in 0..<len] rng.next(65, 122);`

And something similiar happens with `isLoweredForall` (if you want I can rerun paratests to get the exact erroring case).